### ISSUE TITLE
Breaking changes with latest Firebase Auth update

### ISSUE THREAD

**Issue by CHR15TOV** [delay/flutter_starter#2]
Hey Jeff,
Thanks for sharing your work and all was working well until Tuesday!
There are breaking changes with the newest version of Firebase Auth released a few days ago. I've managed to solve the other problems and just this one remains.

in the auth_controller.dart file you declare
`final FirebaseAuth _auth = FirebaseAuth.instance; `    
and the getter
`Future<User> get getUser => _auth.currentUser();`       <===== this changed from `<FirebaseUser>` to just `<User>`
and
`Stream<User> get user => _auth.authStateChanges();`  <===== this changed from `_auth.onAuthStateChanged()`

the `currentUser` method has changed and this now fails due to 'the expression doesn't evaluate to a function, so it can't be invoked.' 

BREAKING: Accessing the current user via `currentUser()` is now synchronous via the currentUser getter is the problem

I've updated the main.dart file to look like this: 
Future main() async {
WidgetsFlutterBinding.ensureInitialized();
  Get.put<AuthController>(AuthController());
  Get.put<ThemeController>(ThemeController());
  Get.put<LanguageController>(LanguageController());
`await Firebase.initializeApp();`                                             <=====  added to ensure that Firebase is initialized prior to runApp
await GetStorage.init();   
  runApp(MyApp());
}

Any thoughts on how to fix this synchronous issue ?
Thanks C

**Comment by christheoalex** [comment]
Seems as there are still problems. Sign up is not possible with a fresh firestore connection.

**Comment by delay** [comment]
What error message are you receiving?

**Comment by CHR15TOV** [comment]
Might be the change to the onClose() method in the auth_controller which changed recently:

Changed mine to:
```
@override
  Future<void> onClose() async {                                        //  dispose of controllers
    nameController?.dispose();
    emailController?.dispose();
    passwordController?.dispose();
    super.onClose();
  }
```

**Comment by christheoalex** [comment]
There was an update method used within the signup call, but the document wasn't created beforehand. I've just replaced it with set.

**Comment by jonataslaw** [comment]
Hi Jeff, I updated Get and GetStorage
GetStorage working, 
onClose accept "void" and "Future" now

**Comment by delay** [comment]
Thanks I updated to the latest packages.  I also fixed the problem with creating a new account. @christheoalex thanks for the info.  I made a new function _createUserFirestore for the firestore set because I need to use the _updateUserFirestore in another function if the user changes his info.  I think I didn't double check when Firestore made all of there api changes a few weeks ago and I broke the registration on that update.
